engine/src/test/java/com/redhat/cloud/notifications/transformers/BaseTransformerTest.java

```diff
@@ -1,36 +1,114 @@
+
+    /**
+     * Sets up the context and a list with one event for the tests.
+     */
+    @BeforeAll
+    static void setUp() {
+        final Context context = new Context();
+
+        context.setAdditionalProperty(FIXTURE_CONTEXT_ADDITIONAL_PROPERTY, FIXTURE_CONTEXT_ADDITIONAL_PROPERTY_VALUE);
+
+        FIXTURE_CONTEXT = context;
+
+        final List<Event> events = new ArrayList<>();
+        final Event event = new Event();
+
+        final Metadata metadata = new Metadata();
+        metadata.setAdditionalProperty(FIXTURE_METADATA_ADDITIONAL_PROPERTY, FIXTURE_METADATA_ADDITIONAL_PROPERTY_VALUE);
+        event.setMetadata(metadata);
 
-        final JsonObject jsonObject = testee.toJsonObject(action);
+        final Payload payload = new Payload();
+        payload.setAdditionalProperty(FIXTURE_PAYLOAD_ADDITIONAL_PROPERTY, FIXTURE_PAYLOAD_ADDITIONAL_PROPERTY_VALUE);
+        event.setPayload(payload);
 
-        assertEquals("someOrgId", jsonObject.getString("org_id"));
+        events.add(event);
+        FIXTURE_EVENTS = events;
     }
 
+    /**
+     * Tests that a proper JSON payload is generated from a fully populated {@link Action}.
+     */
     @Test
-    void shouldNotRaiseAnExceptionWhenOrgIdIsMissing() {
+    void toJsonObjectTest() {
         Action action = new Action();
-        action.setTimestamp(LocalDateTime.of(2022, 12, 24, 12, 0));
 
-        final JsonObject jsonObject = testee.toJsonObject(action);
+        action.setAccountId(FIXTURE_ACCOUNT_ID);
+        action.setApplication(FIXTURE_APPLICATION);
+        action.setBundle(FIXTURE_BUNDLE);
+        action.setContext(FIXTURE_CONTEXT);
+        action.setEventType(FIXTURE_EVENT_TYPE);
+        action.setEvents(FIXTURE_EVENTS);
+        action.setOrgId(FIXTURE_ORG_ID);
+        action.setTimestamp(FIXTURE_TIMESTAMP);
+
+        // Call the function under test.
+        JsonObject result = this.baseTransformer.toJsonObject(action);
+
+        // Assert the values.
+        assertEquals(FIXTURE_ACCOUNT_ID, result.getString(BaseTransformer.ACCOUNT_ID), "the account id isn't the same");
+        assertEquals(FIXTURE_APPLICATION, result.getString(BaseTransformer.APPLICATION), "the application isn't the same");
+        assertEquals(FIXTURE_BUNDLE, result.getString(BaseTransformer.BUNDLE), "the bundle isn't the same");
+        assertEquals(FIXTURE_EVENT_TYPE, result.getString(BaseTransformer.EVENT_TYPE), "the event type isn't the same");
+
+        // Assert that the keys that contain inner JSON arrays or objects are present.
+        assertTrue(result.containsKey(BaseTransformer.CONTEXT), "the context is missing from the resulting JSON object");
+        assertTrue(result.containsKey(BaseTransformer.EVENTS), "the events is missing from the resulting JSON object");
+        assertTrue(result.containsKey(BaseTransformer.ORG_ID), "the org id is missing from the resulting JSON object");
```

Comment:
This could be removed since there's a value check below.
```suggestion
```

Reply:
Done!